src/lib/Save.js

```diff
@@ -0,0 +1,32 @@
+export default class Save {
+  constructor(logs) {
+    this._logs = logs;
+  }
+
+  downloadFile(blob, filename) {
+    const url = URL.createObjectURL(blob);
+
+    function listener(result) {
+      if (result.state.current === 'complete') {
+        setTimeout(() => {
+          URL.revokeObjectURL(url);
+        }, 100);
+        browser.downloads.onChanged.removeListener(listener);
+      }
+    }
+
+    browser.downloads.onChanged.addListener(listener);
+
+    browser.downloads.download({
+      url,
+      filename,
+    });
+  }
+
+  saveAsJSON() {
```

Comment:
from an API design perspective, could the API method meant to save the logs be an async function? (one that return a promise that is going to be resolved when the download has been completed, or rejected if it did fail)
That should likely make it easier to use by the caller.

Reply:
Yeah, I agree. :-)